recipes/apr/all/conanfile.py

```diff
@@ -67,6 +62,16 @@ def _configure_cmake(self):
     def _configure_autotools(self):
         if self._autotools:
             return self._autotools
+        if (self.settings.compiler == "apple-clang" and
+            tools.Version(self.settings.compiler.version) == "12" and
+            self.version == "1.7.0"):
+            if tools.cross_building(self):
+                raise ConanInvalidConfiguration("apr cannot be cross compiled due to runtime checks")
```

Comment:
These kind of checks need to happen in `validate`.
For c3i, I think you need to replace
```python
if tools.cross_building(self):
```
with
```python
if hasattr(self, "settings_build") and tools.cross_building(self):  # to allow c3i to build Macos recipes.
```

Reply:
I personally don't like adding these workarounds specific for the (current) c3i design
@jgsogo WDYT?

Reply:
It's not a workaround, cross-compilation doesn't seem to work currently in this recipe. The message is misleading, that's all.

Reply:
@SpaceIm 
I am talking about this line:
```
if hasattr(self, "settings_build") and tools.cross_building(self):
```
the first part of the condition, specifically - it seems to be a workaround we're copying across the recipes now.
the reason, apparently, `tools.cross_building` may report different results depends on where do you execute it.
e.g. although not using actual cross-building, we call `conan info` on linux machine for Windows profile in CCI, and `tools.cross_building` returns true for that case; but it would return false if running on Windows machine.
so, sometimes it returns different output for the same instance.
thus, leaving the condition as is:
```
if tools.cross_building(self):
```
practically means CCI may reject some of Windows and Linux profiles, although they aren't actually require any cross-building. that's the reason we put extra part of the condition (`hasattr(self, "settings_build")`), which shouldn't be normally needed in the ideal case.
perhaps, we could switch to the new implementation in 1.38 which shouldn't have this issue:
https://github.com/conan-io/conan/blob/1.38.0/conan/tools/cross_building.py#L2

Reply:
Understood. At least it doesn't hurt.